cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java

```diff
@@ -62,6 +62,9 @@
   public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreException, SQLException {
     // create filesystem directory
     super.create(namespaceMeta);
+    if (NamespaceId.DEFAULT.equals(namespaceMeta.getNamespaceId())) {
```

Comment:
in fact, user may not want default namespace at all. There should be a configuration to tell whether `DEFAULT` will be created automatically. Otherwise than that, default namespace should act just like a regular namespace.